workflow/validate/validate.go

```diff
@@ -214,15 +254,42 @@ func ValidateWorkflow(wftmplGetter templateresolution.WorkflowTemplateNamespaced
 	return wfConditions, nil
 }
 
+func ValidateFieldsWithWFTRef(v interface{}) error {
+	val := reflect.ValueOf(v)
```

Comment:
Why do we need to use `reflect` here at all if the only time this function is called is with a `WorkflowSpec` type?

This method is not only hard to read, but it's also susceptible to runtime errors and is more expensive to run.

Reply:
Is there any other recommendation to validate the fields in a struct?  I tried with Json marshal and Structs to Map options But WorkflowSpec Struct has few primitive data types. it didn't work.